### PR TITLE
2356 Added endpoint for uploading contributor pictures into S3 bucket

### PR DESCRIPTION
**Pull request by mironovandrei**
#2356 
Added endpoint for uploading contributor pictures into S3 bucket

### CODE REVIEW

File: src/main/kotlin/com/epam/brn/controller/CloudController.kt [hunk 1]
```diff
@@ -71,4 +71,15 @@ class CloudController(
         cloudUploadService.uploadUnverifiedPictureFile(multipartFile)
         return ResponseEntity(HttpStatus.CREATED)
     }
+
+    @PostMapping(value = ["/upload/contributor/picture"], consumes = [ MediaType.MULTIPART_FORM_DATA_VALUE ])
+    @ApiOperation("Upload picture of contributor")
+    @RolesAllowed(BrnRole.ADMIN)
+    fun uploadContributorPicture(
+        @RequestParam(value = "file") multipartFile: MultipartFile
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/controller/CloudController.kt:79]: there we should get as parameter contributorId
and later befor saving use contributor.NameEn as file name
return: full file url

Comment (mironovandrei) [reply src/main/kotlin/com/epam/brn/controller/CloudController.kt:79]: yes

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -78,12 +78,15 @@ brn.picture.file.default.path=pictures/%s.jpg
 brn.resources.default-pictures.path=pictures/
 # The path for upload unverified picture to cloud
 brn.resources.unverified-pictures.path=pictures/unverified
+# The path for upload contributor pictures
+brn.resources.contributor-pictures.path=brn/pictures/contributors
```
Comment (ElenaSpb) [inline src/main/resources/application.properties:82]: on s3 we have bucket brnup
and folders /pictures/contributors
for contributor pictures

File: src/main/kotlin/com/epam/brn/service/CloudUploadService.kt [hunk 3]
```diff
@@ -23,44 +22,71 @@ class CloudUploadService(
...
+        } else if (cloudService.isFileExist(unverifiedPicturesPath, fileName)) {
+            throw IllegalArgumentException("File \"$fullFileName\" is exist in cloud path \"$unverifiedPicturesPath\"")
+        }
+
+        cloudService.uploadFile(unverifiedPicturesPath, fullFileName, multipartFile.inputStream)
     }
 
-    fun uploadUnverifiedPictureFile(multipartFile: MultipartFile) {
+    fun uploadContributorPicture(multipartFile: MultipartFile): String {
         val fileName = FilenameUtils.getBaseName(multipartFile.originalFilename)
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/service/CloudUploadService.kt:64]: file name should be input parameter as we get it in end-point.
?

File: src/main/resources/application.properties (hunk 2 above)
Comment (mironovandrei) [reply src/main/resources/application.properties]: fixed

File: src/main/kotlin/com/epam/brn/service/CloudUploadService.kt (hunk 3 above)
Comment (mironovandrei) [reply src/main/kotlin/com/epam/brn/service/CloudUploadService.kt]: added parameter

**Review by ElenaSpb — APPROVED** [review]

**Pull request merged**